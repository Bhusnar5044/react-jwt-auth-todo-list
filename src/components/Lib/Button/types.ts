export type Props = {
    type?: 'button' | 'submit' | 'reset';
    disabled?: boolean;
    onClick?: () => void;
    loading?: boolean;
};
