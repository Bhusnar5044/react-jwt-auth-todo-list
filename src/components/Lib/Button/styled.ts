import { styled } from '@utils';
import { Props } from './types';

export const ButtonContainer = styled.button<Props>`
    display: inline-block;
    outline: 0;
    cursor: pointer;
    text-align: center;
    border: 0;
    padding: 7px 16px;
    min-height: 36px;
    min-width: 36px;
    color: #ffffff;
    background: #008060;
    border-radius: 4px;
    font-weight: 500;
    font-size: 14px;
    box-shadow: rgba(0, 0, 0, 0.05) 0px 1px 0px 0px, rgba(0, 0, 0, 0.2) 0px -1px 0px 0px inset;
    &.disabled {
        opacity: 0.7;
        pointer-events: none;
    }
    :hover {
        background: #006e52;
    }
`;
