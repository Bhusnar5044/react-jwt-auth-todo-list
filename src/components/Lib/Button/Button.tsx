import { FC, PropsWithChildren, memo } from 'react';
import { ButtonContainer } from './styled';
import { Props } from './types';

export const Button: FC<PropsWithChildren & Props> = memo(({ children, disabled, ...props }) => {
    return (
        <ButtonContainer className={disabled ? 'disabled' : ''} {...props}>
            {children}
        </ButtonContainer>
    );
});

Button.displayName = 'Button';
Button.defaultProps = {
    type: 'button',
};
