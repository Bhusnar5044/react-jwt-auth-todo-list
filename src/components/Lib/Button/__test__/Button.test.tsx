import { render, screen } from '@test-utils';
import { Button } from '../Button';

describe('Button', () => {
    it('should render properly', () => {
        const { container } = render(<Button />);
        expect(container).toMatchSnapshot();
    });

    it('should render with content', () => {
        render(<Button>Dummy</Button>);
        expect(screen.getByText(/Dummy/i)).toBeInTheDocument();
    });
});
