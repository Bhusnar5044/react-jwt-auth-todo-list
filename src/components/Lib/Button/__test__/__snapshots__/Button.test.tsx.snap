// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render properly 1`] = `
.c0 {
  display: inline-block;
  outline: 0;
  cursor: pointer;
  text-align: center;
  border: 0;
  padding: 7px 16px;
  min-height: 36px;
  min-width: 36px;
  color: #ffffff;
  background: #008060;
  border-radius: 4px;
  font-weight: 500;
  font-size: 14px;
  box-shadow: rgba(0, 0, 0, 0.05) 0px 1px 0px 0px,rgba(0, 0, 0, 0.2) 0px -1px 0px 0px inset;
}

.c0 :hover {
  background: #006e52;
}

<div>
  <button
    class="c0"
    type="button"
  />
</div>
`;
